src/ArmTemplates/Extractor/EntityExtractors/PolicyExtractor.cs

```diff
@@ -194,14 +241,23 @@ public async Task<Template<PolicyTemplateResources>> GenerateGlobalServicePolicy
             string baseFilesDirectory,
             string policyFileName)
         {
-            string policyXMLContent = policyTemplateResource.Properties.PolicyContent;
             var policiesDirectory = Path.Combine(baseFilesDirectory, PoliciesDirectoryName);
+            var fullPolicyPathWithName = Path.Combine(policiesDirectory, policyFileName);
+            FileWriter.CreateFolderIfNotExists(policiesDirectory); // creating <files-root>/policies
+
+            var policyXMLContent = policyTemplateResource.Properties.PolicyContent;
+
+            if (this.policyPathContentCache.ContainsKey(fullPolicyPathWithName))
+            {
+                this.logger.LogError("Policy content already exists at {0}. Possibly, policy will be overwritten!", fullPolicyPathWithName);
```

Comment:
How can this ever happen? If it can't, I'd just get rid of it. If it can
```suggestion
                this.logger.LogError("Policy content already exists in '{0}' and will be overwritten!", fullPolicyPathWithName);
```

Reply:
Again - we didn't refactor the whole extractor still and I am curios if we can meet this because of passing policyTemplates all around the program.

Reply:
Fair enough.
Can you update the message though? Or why do you say "Possibly"?

Reply:
Done it, blame the github for not showing it in the review page :D

![image](https://user-images.githubusercontent.com/31598696/159737225-753bcf74-1071-44a8-884f-89306adbe0fd.png)